extensions/mapreduce/src/main/java/org/radargun/stages/mapreduce/MapReduceStage.java

```diff
@@ -121,20 +126,31 @@
    @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
    private MapReducer<KOut, VOut, R> mapReducer;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Clustered clustered;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private CacheInformation cacheInformation;
 
-   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   @InjectTrait
    private BasicOperations basicOperations;
 
    @InjectTrait(dependency = InjectTrait.Dependency.OPTIONAL)
    private Iterable iterable;
 
    private boolean supportsResultCacheName = false;
 
+   @Init
+   public void validate() {
```

Comment:
@rvansa and @mcimbora I have no problem removing the ability to store the results in a cache. I don't think it is ever used in any of our other benchmarks. However, I do see a distinction about where the input data comes from. The Spark plugin does't need to reply on the cache, because the Spark connector uses HotRod to get the input data from a RemoteCache running in an Infinispan/JDG server. Map/Reduce's input data is coming from a cache, so I think it still needs to rely on the cache, right? Or am I missing some other way that the input data can be retrieved? 
I'm also not sure how the same job can be used for Infinispan and Spark unless they support the same API, and I thought that's what the distributed streams API allowed?